internal/retryafter.go

```diff
@@ -23,11 +23,19 @@ func ExtractRetryAfterHeader(resp *http.Response) OptionalDuration {
 		resp.StatusCode == http.StatusTooManyRequests {
 		retryAfter := strings.TrimSpace(resp.Header.Get(retryAfterHTTPHeader))
 		if retryAfter != "" {
+			// Try to parse retryAfterHTTPHeader as delay-seconds
 			retryIntervalSec, err := strconv.Atoi(retryAfter)
 			if err == nil {
 				retryInterval := time.Duration(retryIntervalSec) * time.Second
 				return OptionalDuration{Defined: true, Duration: retryInterval}
 			}
+			// Try to parse retryAfterHTTPHeader as HTTP-date
+			// See https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.3
+			t, err := time.Parse(time.RFC1123, retryAfter)
```

Comment:
The HTTP RFC7231 defines 3 different formats of HTTP-date that receivers MUST support. One of the formats is RFC5322. This uses RFC1123 which does not seem to be what RFC7231 requires.